### PR TITLE
Modal: small refactor

### PR DESCRIPTION
**Pull request by Zombiefruit**
Adding some labels re: accessibility.

### CODE REVIEW

File: test/components/modal.spec.tsx [hunk 1]
```diff
@@ -131,7 +131,7 @@ describe('<Modal />', () => {
         await waitFor(() => expect(onRequestClose).to.have.been.calledWithMatch({source: 'backdrop'}));
     });
 
-    it('calls onRequestClose with source equal to children when the child is clicked', async function() {
+    it('calls onRequestClose with source equal to the data-slot of the clicked element', async function() {
```
Comment (tobich) [inline test/components/modal.spec.tsx]: `data-slot="child"` or `data-slot="children"`? I see inconsistency

Comment (Zombiefruit) [reply test/components/modal.spec.tsx]: Now the test is simply checking that the proper data-slot is returned for whatever child element is clicked. There's no longer a wrapper with data-slot equal to children

File: demo/components/modal-demo.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ export class ModalDemo extends React.Component<{}, ModalDemoState> {
             <div>
                 <button data-automation-id="MODAL_BUTTON" onClick={this.toggleOpen}>Open The Modal!</button>
                 <Modal className="root" isOpen={this.state.isOpen} onRequestClose={this.onModalClick}>
-                    <div role="children" className="content">🌌</div>
+                    <div data-slot="children" className="content">🌌</div>
```
Comment (liorzisman) [inline demo/components/modal-demo.tsx]: wasnt the role important for the accessibility?

File: src/components/modal/modal.st.css [hunk 3]
```diff
@@ -9,7 +9,7 @@
     background: rgba(0, 0, 0, 0.05);
 }
 
-.children {
+.body {
```
Comment (JoMarton) [inline src/components/modal/modal.st.css]: I don't know if this might cause any problem or confusion but naming a class `body` seems weird to me

**Pull request merged**